integrations/openrouter/src/haystack_integrations/components/generators/openrouter/chat/chat_generator.py

```diff
@@ -0,0 +1,193 @@
+                considers the results of the tokens with top_p probability mass. So 0.1 means only the tokens
+                comprising the top 10% probability mass are considered.
+            - `stream`: Whether to stream back partial progress. If set, tokens will be sent as data-only server-sent
+                events as they become available, with the stream terminated by a data: [DONE] message.
+            - `safe_prompt`: Whether to inject a safety prompt before all conversations.
+            - `random_seed`: The seed to use for random sampling.
+        :param tools:
+            A list of tools for which the model can prepare calls.
+        :param timeout:
+            The timeout for the OpenRouter API call.
+        :param extra_headers:
+            Extra headers for the OpenRouter API call.
+            For more details, see OpenRouter [docs](https://openrouter.ai/docs/quickstart).
+        :param max_retries:
+            Maximum number of retries to contact OpenAI after an internal error.
+            If not set, it defaults to either the `OPENAI_MAX_RETRIES` environment variable, or set to 5.
+
+        """
+        super(OpenRouterChatGenerator, self).__init__(  # noqa: UP008
+            api_key=api_key,
+            model=model,
+            streaming_callback=streaming_callback,
+            api_base_url=api_base_url,
+            generation_kwargs=generation_kwargs,
+            tools=tools,
+            timeout=timeout,
+            max_retries=max_retries,
+        )
+        self.extra_headers = extra_headers
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serialize this component to a dictionary.
+
+        :returns:
+            The serialized component as a dictionary.
+        """
+        callback_name = serialize_callable(self.streaming_callback) if self.streaming_callback else None
+
+        # if we didn't implement the to_dict method here then the to_dict method of the superclass would be used
+        # which would serialiaze some fields that we don't want to serialize (e.g. the ones we don't have in
+        # the __init__)
+        # it would be hard to maintain the compatibility as superclass changes
+        return default_to_dict(
+            self,
+            model=self.model,
+            streaming_callback=callback_name,
+            api_base_url=self.api_base_url,
+            generation_kwargs=self.generation_kwargs,
+            api_key=self.api_key.to_dict(),
+            tools=[tool.to_dict() for tool in self.tools] if self.tools else None,
+            extra_headers=self.extra_headers,
+            timeout=self.timeout,
+            max_retries=self.max_retries,
+        )
+
+    def _prepare_api_call(
+        self,
+        *,
+        messages: List[ChatMessage],
+        streaming_callback: Optional[StreamingCallbackT] = None,
+        generation_kwargs: Optional[Dict[str, Any]] = None,
+        tools: Optional[List[Tool]] = None,
```

Comment:
Let's add support for `Toolset` here as well
```suggestion
        tools: Optional[Union[List[Tool], Toolset]] = None,
```